### PR TITLE
Draft database

### PR DESCRIPTION
**Pull request by antaranyan**
Setup DB with Knex (using migrations) and implement some methods

### CODE REVIEW

File: packages/db/src/methods/helpers/deleteOneFrom.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+const R = require('ramda')
+
+const getOneFrom = (knex, tableName, selector) =>
```
Comment (mushegha) [inline packages/db/src/methods/helpers/deleteOneFrom.js]: fix naming

Comment (antaranyan) [inline packages/db/src/methods/helpers/deleteOneFrom.js:3]:
```suggestion
const deleteOneFrom = (knex, tableName, selector) =>
```

File: packages/db/src/methods/helpers/deleteOneFrom.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+  knex(tableName)
+    .where(selector)
+    .limit(1)
+    .del()
+    .then(R.head) //
+
+module.exports = R.curry(getOneFrom)
```
Comment (antaranyan) [inline packages/db/src/methods/helpers/deleteOneFrom.js:10]:
```suggestion
module.exports = R.curry(deleteOneFrom)
```

File: packages/db/src/methods/helpers/deleteOneFrom.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+const R = require('ramda')
+
+const getOneFrom = (knex, tableName, selector) =>
+  knex(tableName)
+    .where(selector)
+    .limit(1)
+    .del()
+    .then(R.head) //
```
Comment (antaranyan) [inline packages/db/src/methods/helpers/deleteOneFrom.js:8]: Not useful response, should indicate if deleted or not, and maybe return deleted row

File: packages/db/src/methods/removeWorkspaceById.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+const R = require('ramda')
+
+const { deleteOneFrom } = require('./helpers')
+
+/**
+ *
+ */
+
+function removeWorkspaceById (knex, id) {
+  return deleteOneFrom(knex, 'workspaces', { id })
```
Comment (antaranyan) [inline packages/db/src/methods/removeWorkspaceById.js:10]: consider defining table names somewhere else as constant and use like `tableNames.workspaces`.

same goes for all table names

File: packages/db/src/migrations/20200311015053_create_users_table.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
+const TABLE_NAME = 'users'
+
+exports.up = knex => {
+  return knex.schema
+    .createTable(TABLE_NAME, table => {
+      table.increments('id').unsigned().primary()
```
Comment (antaranyan) [inline packages/db/src/migrations/20200311015053_create_users_table.js:6]: sequential id's are easy but not considered a best practice. Some third party could derive a lot from it which is not desired. consider using some UUID or hash id's on API side.

File: packages/db/test/helper-methods.spec.js [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+  // assert effect
+
+  await knex('users')
+    .where({ id: user.id })
+    .then(res => {
+      t.deepEqual(res, [user])
+    })
+})
+
+test('findOneFrom', async t => {
```
Comment (antaranyan) [inline packages/db/test/helper-methods.spec.js:56]: missing cases
- find exactly one from multiple matches
- no match

File: packages/db/ava.config.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
+const files = [
+  'test/**/*.spec.js',
+  'test/**/*.test.js'
+]
+
+const ignoredByWatcher = [
+  'tmp/**'
```
Comment (antaranyan) [inline packages/db/ava.config.js:7]: unnecessary

File: packages/db/config/development.js [hunk 8]
```diff
@@ -0,0 +1,34 @@
+const { resolve: resolvePath } = require('path')
+const { sync: getRootDir } = require('pkg-dir')
+
+//
+
+const ROOT_DIR = getRootDir()
+
+const DB_FILENAME = '/tmp/slaq-dev.sqlite3'
```
Comment (antaranyan) [inline packages/db/config/development.js:8]: allow passing via environment variable

**Pull request merged**